src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogView.java

```diff
@@ -0,0 +1,120 @@
+import org.jabref.gui.util.ControlHelper;
+import org.jabref.gui.util.IconValidationDecorator;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.shared.DBMSType;
+
+import com.airhacks.afterburner.views.ViewLoader;
+import de.saxsys.mvvmfx.utils.validation.visualization.ControlsFxVisualizer;
+import org.fxmisc.easybind.EasyBind;
+
+public class SharedDatabaseLoginDialogView extends BaseDialog<Void> {
+
+    private final JabRefFrame frame;
+
+    @FXML private ComboBox<DBMSType> cmbDbType;
+    @FXML private TextField tbHost;
+    @FXML private TextField tbDb;
+    @FXML private TextField tbPort;
+    @FXML private TextField tbUser;
+    @FXML private PasswordField tbPwd;
+    @FXML private CheckBox chkRememberPassword;
+    @FXML private TextField tbFolder;
+    @FXML private Button btnBrowse;
+    @FXML private CheckBox chkAutosave;
+    @FXML private ButtonType connectButton;
+
+    @Inject private DialogService dialogService;
+
+    private final Button btnConnect;
+
+    private SharedDatabaseLoginDialogViewModel viewModel;
+    private final ControlsFxVisualizer visualizer = new ControlsFxVisualizer();
+
+    public SharedDatabaseLoginDialogView(JabRefFrame frame) {
+        this.frame = frame;
+        this.setTitle(Localization.lang("Connect to shared database"));
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        ControlHelper.setAction(connectButton, this.getDialogPane(), event -> openDatabase());
+        btnConnect = (Button) this.getDialogPane().lookupButton(connectButton);
+        setLoadingConnectButtonText();
+        btnConnect.disableProperty().bind(viewModel.formValidation().validProperty().not());
+
+    }
+
+    @FXML
+    private void openDatabase() {
+        viewModel.openDatabase();
+    }
+
+    @FXML
+    private void initialize() {
+        visualizer.setDecoration(new IconValidationDecorator());
+
+        viewModel = new SharedDatabaseLoginDialogViewModel(frame, dialogService);
+        cmbDbType.itemsProperty().bind(viewModel.dbmstypeProperty());
+        cmbDbType.getSelectionModel().select(0);
+
+        tbDb.textProperty().bindBidirectional(viewModel.dbProperty());
+        tbHost.textProperty().bindBidirectional(viewModel.hostProperty());
+        tbUser.textProperty().bindBidirectional(viewModel.userProperty());
+        tbPwd.textProperty().bindBidirectional(viewModel.passwordProperty());
+        tbPort.textProperty().bindBidirectional(viewModel.portProperty());
+        cmbDbType.valueProperty().bindBidirectional(viewModel.selectedDbmstypeProperty());
+
+        tbFolder.textProperty().bindBidirectional(viewModel.folderProperty());
+        btnBrowse.disableProperty().bind(chkAutosave.selectedProperty().not());
+        tbFolder.disableProperty().bind(chkAutosave.selectedProperty().not());
+        chkAutosave.selectedProperty().bindBidirectional(viewModel.autosaveProperty());
+
+        Platform.runLater(() -> {
```

Comment:
Please add a comment that this is a workaround and add a link to the issue in controlsfx